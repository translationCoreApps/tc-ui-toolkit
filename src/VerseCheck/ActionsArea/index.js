import React from 'react';
import {Glyphicon} from 'react-bootstrap';
import isEqual from 'deep-equal';
import Checkbox from '@material-ui/core/Checkbox';
import { withStyles } from '@material-ui/core/styles';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import ReactTooltip from 'react-tooltip';
// components
import Bookmark from '../../Bookmark';
// css
import './ActionsArea.styles.css';

const styles = {
  formControl: {
    margin: '0'
  },
  label: {
    color: 'var(--accent-color-dark)',
    fontWeight: "normal",
    fontSize: 14,
  },
  checkBoxRoot: {
    padding: '12px 5px',
    color: 'var(--accent-color-dark)',
    '&$checked': {
      color: 'var(--accent-color-dark)',
    },
  },
  checked: {},
};


const isSelectionsSaveDisable = (localNothingToSelect, nothingToSelect, newSelections, selections) => {
  console.log('selections', selections);
  if (newSelections.length > 0 || (newSelections.length === 0 && !isEqual(newSelections, selections))) {
    return isEqual(newSelections, selections);
  }

  return localNothingToSelect === nothingToSelect;
};

let ActionsArea = ({
  tags,
  mode,
  actions,
  commentChanged,
  selections,
  newSelections,
  remindersReducer,
  saveSelection,
  cancelSelection,
  clearSelection,
  translate,
  classes,
  localNothingToSelect,
  nothingToSelect,
  toggleNothingToSelect,
}) => {

  const changeModeArea = (
    <div className='actions-area'>
      <Bookmark
        value='bookmark'
        color='primary'
        checked={remindersReducer.enabled}
        label={translate('bookmark')}
        onChange={actions.toggleReminder} />
      <div style={{display: "flex", marginLeft: 'auto'}}>
        <button
          style={{width: "140px", marginRigth: "5px"}}
          className='btn-second'
          onClick={actions.changeMode.bind(this, 'select')}
        >
          <Glyphicon glyph='ok' style={{marginRight: '10px'}} />
          {translate("select")}
        </button>
        <button
          style={{width: "140px", marginRigth: "5px"}}
          className='btn-second'
          onClick={actions.changeMode.bind(this, 'edit')}
        >
          <Glyphicon glyph='pencil' style={{marginRight: '10px'}} />
          {translate("edit_verse")}
        </button>
        <button
          style={{width: "140px"}}
          className='btn-second'
          onClick={actions.changeMode.bind(this, 'comment')}
        >
          <Glyphicon glyph='comment' style={{marginRight: '10px'}} />
          {translate("comment")}
        </button>
      </div>
    </div>
  );

  const confirmEditVerseArea = (
    <div className='actions-area'>
      <button className='btn-second'
        onClick={actions.cancelEditVerse.bind(this)}
      >
        {translate("cancel")}
      </button>
      <button className='btn-prime'
        disabled={!tags.length}
        onClick={actions.saveEditVerse.bind(this)}
      >
        <Glyphicon glyph='ok' style={{marginRight: '10px'}} />
        {translate("save")}
      </button>
    </div>
  );

  const confirmCommentArea = (
    <div className='actions-area'>
      <button className='btn-second'
        onClick={actions.cancelComment.bind(this)}
      >
        {translate("cancel")}
      </button>
      <button className='btn-prime'
        disabled={!commentChanged}
        onClick={actions.saveComment.bind(this)}
      >
        <Glyphicon glyph='ok' style={{marginRight: '10px'}} />
        {translate("save")}
      </button>
    </div>
  );
  const confirmSelectionArea = (
    <div className='selection-actions-area'>
      <div
        data-tip={translate('nothing_to_select_description')}
        data-place="top"
        data-effect="float"
        data-type="dark"
        data-class="selection-tooltip"
        data-delay-hide="100"
        style={{ verticalAlign: 'super', fontSize: '0.8em' }}
      >
        <FormControlLabel
          value="end"
          control={
            <Checkbox
              checked={localNothingToSelect}
              disabled={newSelections && newSelections.length > 0}
              onChange={event => toggleNothingToSelect(event.target.checked)}
              value="nothingToSelect"
              color="primary"
              classes={{
                root: classes.checkBoxRoot,
                checked: classes.checked,
              }}
            />
          }
          label="No selection needed"
          classes={{
            root: classes.formControl,
            label: classes.label
          }}
        />
        <ReactTooltip/>
      </div>
      <div>
        <button
          className='btn-second'
          style={{width: "140px", marginRigth: "5px", alignSelf: 'flex-start'}}
          onClick={cancelSelection.bind(this)}
        >
          {translate("cancel")}
        </button>
        <button
          className='btn-second'
          style={{width: "140px", marginRigth: "5px"}}
          disabled={newSelections.length > 0 ? false : true}
          onClick={clearSelection.bind(this)}
        >
          <Glyphicon glyph='erase' style={{marginRight: '10px'}} />
          {translate("clear_selection")}
        </button>
        <button
          className='btn-prime'
          style={{width: "140px", marginRigth: "5px"}}
          disabled={isSelectionsSaveDisable(localNothingToSelect, nothingToSelect, newSelections, selections)}
          onClick={saveSelection.bind(this)}
        >
          <Glyphicon glyph='ok' style={{marginRight: '10px'}} />
          {translate("save")}
        </button>
      </div>
    </div>
  );

  let modeArea;
  switch (mode) {
    case 'edit':
      modeArea = confirmEditVerseArea;
      break;
    case 'comment':
      modeArea = confirmCommentArea;
      break;
    case 'select':
      modeArea = confirmSelectionArea;
      break;
    case 'default':
      modeArea = changeModeArea;
      break;
    default:
      modeArea = changeModeArea;
  }

  return modeArea;
};

export default withStyles(styles)(ActionsArea);
