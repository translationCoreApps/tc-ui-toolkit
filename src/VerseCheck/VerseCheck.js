/* eslint-disable indent */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ActionsArea from './ActionsArea';
import CheckArea from './CheckArea';
import SaveArea from './SaveArea';
import DialogComponent from './DialogComponent';
import IconIndicators from './IconIndicators';

import './VerseCheck.styles.css';

class VerseCheck extends Component {
  render() {
    const {
      translate,
      alignedGLText,
      verseText,
      unfilteredVerseText,
      mode,
      dialogModalVisibility,
      commentChanged,
      tags,
      verseChanged,
      saveSelection,
      cancelSelection,
      clearSelection,
      handleSkip,
      toggleNothingToSelect,
      localNothingToSelect,
      maximumSelections,
      selections,
      newSelections,
      nothingToSelect,
      verseEdited,
      commentText,
      bookmarkEnabled,
      isVerseInvalidated,
      contextId,
      targetBible,
      handleCloseDialog,
      handleGoToNext,
      handleGoToPrevious,
      handleOpenDialog,
      openAlertDialog,
      changeSelectionsInLocalState,
      toggleReminder,
      changeMode,
      cancelEditVerse,
      saveEditVerse,
      handleComment,
      cancelComment,
      saveComment,
      bookDetails,
      targetLanguageDetails,
      handleTagsCheckbox,
      handleEditVerse,
      hasVerseChanged,
      hasCommentChanged,
      validateSelections,
    } = this.props;

    let titleText;
    let saveArea;

    switch (mode) {
      case 'edit':
        titleText = translate('edit_verse');
        saveArea = <div />;
        break;
      case 'comment':
        titleText = translate('comment');
        saveArea = <div />;
        break;
      case 'select':
        titleText = translate('select');
        saveArea = <div />;
        break;
      default:
        titleText = translate('step2_check');
        saveArea = (
          <SaveArea
            translate={translate}
            selections={selections}
            nothingToSelect={nothingToSelect}
            handleGoToNext={handleGoToNext}
            handleGoToPrevious={handleGoToPrevious}
            handleOpenDialog={handleOpenDialog}
          />
        );
    }

    return (
      <div className='verse-check'>
        <div className='verse-check-flex'>
          <div className='verse-check-card'>
            <div className='title-bar'>
              <span>{titleText}</span>
              <IconIndicators
                verseEdited={verseEdited}
                selections={selections}
                comment={commentText}
                bookmarkEnabled={bookmarkEnabled}
                translate={translate}
                nothingToSelect={nothingToSelect}
                invalidated={isVerseInvalidated}
              />
            </div>
            <CheckArea
              mode={mode}
              tags={tags}
              verseText={verseText}
              unfilteredVerseText={unfilteredVerseText}
              verseChanged={verseChanged}
              comment={commentText}
              newSelections={newSelections}
              selections={selections}
              translate={translate}
              nothingToSelect={nothingToSelect}
              targetLanguageDetails={targetLanguageDetails}
              bookDetails={bookDetails}
              contextId={contextId}
              targetBible={targetBible}// TODO:
              alignedGLText={alignedGLText}
              invalidated={isVerseInvalidated}
              maximumSelections={maximumSelections}
              handleComment={handleComment}
              openAlertDialog={openAlertDialog}
              handleEditVerse={handleEditVerse}
              hasVerseChanged={hasVerseChanged}
              hasCommentChanged={hasCommentChanged}
              handleTagsCheckbox={handleTagsCheckbox}
              validateSelections={validateSelections}
              changeSelectionsInLocalState={changeSelectionsInLocalState}
            />
            <ActionsArea
              mode={mode}
              tags={tags}
              toggleNothingToSelect={toggleNothingToSelect}
              localNothingToSelect={localNothingToSelect}
              nothingToSelect={nothingToSelect}
              commentChanged={commentChanged}
              selections={selections}
              newSelections={newSelections}
              translate={translate}
              bookmarkEnabled={bookmarkEnabled}
              saveSelection={saveSelection}
              cancelSelection={cancelSelection}
              clearSelection={clearSelection}
              toggleReminder={toggleReminder}
              changeMode={changeMode}
              cancelEditVerse={cancelEditVerse}
              saveEditVerse={saveEditVerse}
              cancelComment={cancelComment}
              saveComment={saveComment}
            />
          </div>
          {saveArea}
        </div>
        <DialogComponent
          handleSkip={handleSkip}
          dialogModalVisibility={dialogModalVisibility}
          handleClose={handleCloseDialog}
          translate={translate} />
      </div>
    );
  }
}


VerseCheck.propTypes = {
  translate: PropTypes.func.isRequired,
  mode: PropTypes.string.isRequired,
  tags: PropTypes.array.isRequired,
  selections: PropTypes.array.isRequired,
  newSelections: PropTypes.array.isRequired,
  nothingToSelect: PropTypes.bool.isRequired,
  verseEdited: PropTypes.bool.isRequired,
  commentText: PropTypes.string.isRequired,
  bookmarkEnabled: PropTypes.bool.isRequired,
  isVerseInvalidated: PropTypes.bool.isRequired,
  contextId: PropTypes.object.isRequired,
  targetBible: PropTypes.object.isRequired,
  bookDetails: PropTypes.object.isRequired,
  targetLanguageDetails: PropTypes.object.isRequired,
  alignedGLText: PropTypes.string.isRequired,
  commentChanged: PropTypes.bool.isRequired,
  verseChanged: PropTypes.bool.isRequired,
  verseText: PropTypes.string.isRequired,
  unfilteredVerseText: PropTypes.string.isRequired,
  dialogModalVisibility: PropTypes.bool.isRequired,
  localNothingToSelect: PropTypes.bool.isRequired,
  maximumSelections: PropTypes.number.isRequired,
  handleCloseDialog: PropTypes.func.isRequired,
  handleGoToNext: PropTypes.func.isRequired,
  handleGoToPrevious: PropTypes.func.isRequired,
  handleOpenDialog: PropTypes.func.isRequired,
  openAlertDialog: PropTypes.func.isRequired,
  toggleReminder: PropTypes.func.isRequired,
  changeMode: PropTypes.func.isRequired,
  cancelEditVerse: PropTypes.func.isRequired,
  saveEditVerse: PropTypes.func.isRequired,
  handleComment: PropTypes.func.isRequired,
  cancelComment: PropTypes.func.isRequired,
  saveComment: PropTypes.func.isRequired,
  toggleNothingToSelect: PropTypes.func.isRequired,
  saveSelection: PropTypes.func.isRequired,
  cancelSelection: PropTypes.func.isRequired,
  clearSelection: PropTypes.func.isRequired,
  handleSkip: PropTypes.func.isRequired,
  handleEditVerse: PropTypes.func.isRequired,
  hasVerseChanged: PropTypes.func.isRequired,
  hasCommentChanged: PropTypes.func.isRequired,
  validateSelections: PropTypes.func.isRequired,
  handleTagsCheckbox: PropTypes.func.isRequired,
  changeSelectionsInLocalState: PropTypes.func.isRequired,
};

VerseCheck.defaultProps = {
  translate: key => key,
  verseText: '',
  unfilteredVerseText: '',
  commentText: '',
};

export default VerseCheck;
