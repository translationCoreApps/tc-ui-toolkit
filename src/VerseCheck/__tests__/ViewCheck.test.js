/* eslint-env jest */
import React from 'react';
import VerseCheck from '../VerseCheck';
import renderer from 'react-test-renderer';

const mock_translate = (text) => (text);
const base_props = require('./fixtures/project/loadedProjectShortened');
let currentInvalidated = false;
let currentEdited = false;

describe('VerseCheck component:', () => {
  currentInvalidated = false;
  currentEdited = false;

  test('Integrated View test', () => {
    // given
    const props = getBasePropertiesAndMockActions();
    currentInvalidated = false;
    currentEdited = false;

    // when
    const component = renderer.create(
        <VerseCheck {...props} />
    );

    // then
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('Integrated View test with verseEdit', () => {
    // given
    const props = getBasePropertiesAndMockActions();
    currentInvalidated = false;
    currentEdited = true;

    // when
    const component = renderer.create(
        <VerseCheck {...props} />
    );

    // then
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('Integrated View test with invalidated', () => {
    // given
    const props = getBasePropertiesAndMockActions();
    currentInvalidated = true;
    currentEdited = false;

    // when
    const component = renderer.create(
      <VerseCheck {...props} />
    );

    // then
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('Integrated View test with default and invalidated', () => {
    // given
    const props = getBasePropertiesAndMockActions();
    props.mode = ''; // default mode
    currentInvalidated = true;
    currentEdited = false;

    // when
    const component = renderer.create(
      <VerseCheck {...props} />
    );

    // then
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('Integrated View test - edit mode', () => {
    // given
    const props = getBasePropertiesAndMockActions();
    currentInvalidated = false;
    currentEdited = false;
    props.mode = "edit";

    // when
    const component = renderer.create(
      <VerseCheck {...props} />
    );

    // then
    expect(component.toJSON()).toMatchSnapshot();
  });
});

//
// helpers
//

function addMockActions(props) {
  props.actions = {
    handleGoToNext: () => jest.fn(),
    handleGoToPrevious: () => jest.fn(),
    handleOpenDialog: () => jest.fn(),
    handleCloseDialog: () => jest.fn(),
    skipToNext: () => jest.fn(),
    skipToPrevious: () => jest.fn(),
    changeSelectionsInLocalState: () => jest.fn(),
    changeMode: () => jest.fn(),
    handleComment: () => jest.fn(),
    checkComment: () => jest.fn(),
    cancelComment: () => jest.fn(),
    saveComment: () => jest.fn(),
    handleTagsCheckbox: () => jest.fn(),
    handleEditVerse: () => jest.fn(),
    checkVerse: () => jest.fn(),
    cancelEditVerse: () => jest.fn(),
    saveEditVerse: () => jest.fn(),
    validateSelections: () => jest.fn(),
    toggleReminder: () => jest.fn(),
    openAlertDialog: () => jest.fn(),
    selectModalTab: () => jest.fn(),
    cancelSelection: () => jest.fn(),
    clearSelection: () => jest.fn(),
    saveSelection: () => jest.fn()
  };
  return {
    ...props,
    translate: mock_translate,
    cancelSelection: () => jest.fn(),
    clearSelection: () => jest.fn(),
    saveSelection: () => jest.fn(),
    goToNextOrPrevious: "next",
    findIfVerseEdited: jest.fn(() => {
      return ( currentEdited); }),
    findIfVerseInvalidated: jest.fn(() => {
      return (currentInvalidated); })
  };
}

function getBasePropertiesAndMockActions() {
  // clone properties so we can modify before test
  let props = JSON.parse(JSON.stringify(base_props));
  props = addMockActions(props);
  return props;
}
