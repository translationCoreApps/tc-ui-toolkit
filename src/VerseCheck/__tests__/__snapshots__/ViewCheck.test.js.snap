// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View component Tests Integrated View test 1`] = `
<div
  className="verse-check"
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <div
      className="verse-check-card"
    >
      <div
        className="title-bar"
      >
        <span>
          select
        </span>
        <div
          style={
            Object {
              "display": "flex",
              "justifyContent": "flex-end",
            }
          }
        >
          <span
            className="glyphicon glyphicon-ok"
            style={
              Object {
                "color": "var(--reverse-color)",
                "margin": "0px 20px",
                "opacity": 0.2,
              }
            }
            title="icons.no_selections_found"
          />
          <span
            className="glyphicon glyphicon-pencil"
            style={
              Object {
                "color": "var(--reverse-color)",
                "margin": "0px 20px",
                "opacity": 0.2,
              }
            }
            title="icons.no_verse_edits_found"
          />
          <span
            className="glyphicon glyphicon-comment"
            style={
              Object {
                "color": "var(--reverse-color)",
                "margin": "0px 20px",
                "opacity": 0.2,
              }
            }
            title="icons.no_comments_found"
          />
          <span
            className="glyphicon glyphicon-bookmark"
            style={
              Object {
                "color": "var(--reverse-color)",
                "margin": "0px 20px",
                "opacity": 0.2,
              }
            }
            title="icons.not_bookmarked"
          />
        </div>
      </div>
      <div
        className="check-area"
      >
        <div
          style={
            Object {
              "display": "flex",
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <div
            className="verse-title"
          >
            <div
              className="pane"
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                }
              }
            >
              <span
                className="verse-title-title"
              >
                Fwe
              </span>
              <span
                className="verse-title-subtitle"
              >
                Titus
                 
                2:5
              </span>
            </div>
          </div>
          <div>
            <div
              className="ltr-content"
            >
              <div
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
              >
                <span>
                  Kono ewe, oambe ziyenderera ne ntaero zisepahara. 
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "borderLeft": "1px solid var(--border-color)",
              "display": "flex",
              "flex": 1,
              "justifyContent": "center",
              "overflowY": "auto",
            }
          }
        >
          <div
            style={
              Object {
                "WebkitUserSelect": "none",
                "alignItems": "center",
                "display": "flex",
                "flex": "1",
                "justifyContent": "center",
                "overflow": "auto",
              }
            }
          >
            <div
              className="instructions-area"
            >
              <span>
                please_select
              </span>
              <br />
              <span>
                <strong
                  style={
                    Object {
                      "color": "var(--accent-color)",
                    }
                  }
                >
                  ""
                </strong>
              </span>
              <br />
            </div>
          </div>
        </div>
      </div>
      <div
        className="actions-area"
      >
        <button
          className="btn-second"
          onClick={[Function]}
          style={
            Object {
              "alignSelf": "flex-start",
            }
          }
        >
          cancel
        </button>
        <button
          className="btn-second"
          disabled={true}
          onClick={[Function]}
        >
          <span
            className="glyphicon glyphicon-erase"
            style={
              Object {
                "marginRight": "10px",
              }
            }
          />
          clear_selection
        </button>
        <button
          className="btn-prime"
          disabled={true}
          onClick={[Function]}
        >
          <span
            className="glyphicon glyphicon-ok"
            style={
              Object {
                "marginRight": "10px",
              }
            }
          />
          save
        </button>
      </div>
    </div>
    <div />
  </div>
  <div />
</div>
`;

exports[`View component Tests Integrated View test with invalidated 1`] = `
<div
  className="verse-check"
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <div
      className="verse-check-card"
    >
      <div
        className="title-bar"
      >
        <span>
          select
        </span>
        <div
          style={
            Object {
              "display": "flex",
              "justifyContent": "flex-end",
            }
          }
        >
          <div
            className="glyphicon glyphicon-invalidated"
            style={
              Object {
                "margin": "0px 20px",
              }
            }
          >
            <svg
              height="16"
              style={
                Object {
                  "display": "inline-block",
                  "verticalAlign": "middle",
                }
              }
              viewBox="0 0 475.082 475.082"
              width="16"
            >
              <path
                d="M107.067,317.195c1.713-1.708,2.568-3.898,2.568-6.563c0-2.663-0.855-4.853-2.568-6.571    c-1.714-1.707-3.905-2.562-6.567-2.562H9.135c-2.666,0-4.853,0.855-6.567,2.562C0.859,305.772,0,307.962,0,310.632    c0,2.665,0.855,4.855,2.568,6.563c1.714,1.711,3.905,2.566,6.567,2.566H100.5C103.166,319.766,105.353,318.91,107.067,317.195z"
                style={
                  Object {
                    "fill": "#ffffff",
                  }
                }
              />
              <path
                d="M310.629,109.634c2.669,0,4.859-0.855,6.563-2.568c1.718-1.711,2.574-3.901,2.574-6.567V9.138    c0-2.659-0.856-4.85-2.574-6.565c-1.704-1.711-3.895-2.57-6.563-2.57c-2.662,0-4.853,0.859-6.563,2.57    c-1.711,1.713-2.566,3.903-2.566,6.565v91.361c0,2.666,0.855,4.856,2.566,6.567C305.784,108.779,307.974,109.634,310.629,109.634z    "
                style={
                  Object {
                    "fill": "#ffffff",
                  }
                }
              />
              <path
                d="M118.771,347.184c-2.478,0-4.664,0.855-6.567,2.563l-73.089,73.087c-1.713,1.902-2.568,4.093-2.568,6.567    c0,2.474,0.855,4.664,2.568,6.566c2.096,1.708,4.283,2.57,6.567,2.57c2.475,0,4.665-0.862,6.567-2.57l73.089-73.087    c1.714-1.902,2.568-4.093,2.568-6.57c0-2.471-0.854-4.661-2.568-6.563C123.436,348.039,121.245,347.184,118.771,347.184z"
                style={
                  Object {
                    "fill": "#ffffff",
                  }
                }
              />
              <path
                d="M356.315,127.905c2.283,0,4.473-0.855,6.571-2.565l73.087-73.089c1.707-1.903,2.562-4.093,2.562-6.567    c0-2.475-0.855-4.665-2.562-6.567c-1.91-1.709-4.093-2.568-6.571-2.568c-2.471,0-4.66,0.859-6.563,2.568l-73.087,73.089    c-1.708,1.903-2.57,4.093-2.57,6.567c0,2.474,0.862,4.661,2.57,6.567C351.846,127.05,354.037,127.905,356.315,127.905z"
                style={
                  Object {
                    "fill": "#ffffff",
                  }
                }
              />
              <path
                d="M350.607,193.005c-4-3.999-9.328-7.994-15.988-11.991l-5.14,68.238l78.23,78.508c5.328,5.328,7.987,11.807,7.987,19.417    c0,7.423-2.662,13.802-7.987,19.13l-41.977,41.686c-5.146,5.146-11.608,7.666-19.417,7.566c-7.81-0.1-14.271-2.707-19.411-7.854    l-77.946-78.225l-68.234,5.144c3.999,6.656,7.993,11.988,11.991,15.985l95.362,95.643c15.803,16.18,35.207,24.27,58.238,24.27    c22.846,0,42.154-7.898,57.957-23.695l41.977-41.685c16.173-15.8,24.27-35.115,24.27-57.958c0-22.46-7.994-41.877-23.982-58.248    L350.607,193.005z"
                style={
                  Object {
                    "fill": "#ffffff",
                  }
                }
              />
              <path
                d="M472.518,157.889c-1.711-1.709-3.901-2.565-6.563-2.565h-91.365c-2.662,0-4.853,0.855-6.563,2.565    c-1.715,1.713-2.57,3.903-2.57,6.567c0,2.666,0.855,4.856,2.57,6.567c1.711,1.712,3.901,2.568,6.563,2.568h91.365    c2.662,0,4.853-0.856,6.563-2.568c1.708-1.711,2.563-3.901,2.563-6.567C475.082,161.792,474.226,159.602,472.518,157.889z"
                style={
                  Object {
                    "fill": "#ffffff",
                  }
                }
              />
              <path
                d="M109.348,67.099c5.523-5.14,11.991-7.705,19.417-7.705c7.611,0,14.084,2.663,19.414,7.993l77.943,78.227l68.234-5.142    c-4-6.661-7.99-11.991-11.991-15.987l-95.358-95.643c-15.798-16.178-35.212-24.27-58.242-24.27c-22.841,0-42.16,7.902-57.958,23.7    L28.837,69.955C12.659,85.756,4.57,105.073,4.57,127.912c0,22.463,7.996,41.877,23.982,58.245l95.93,95.93    c3.995,4.001,9.325,7.995,15.986,11.991l5.139-68.521L67.377,147.33c-5.327-5.33-7.992-11.801-7.992-19.417    c0-7.421,2.662-13.796,7.992-19.126L109.348,67.099z"
                style={
                  Object {
                    "fill": "#ffffff",
                  }
                }
              />
              <path
                d="M164.454,365.451c-2.667,0-4.854,0.855-6.567,2.563c-1.711,1.711-2.568,3.901-2.568,6.57v91.358    c0,2.669,0.854,4.853,2.568,6.57c1.713,1.707,3.9,2.566,6.567,2.566c2.666,0,4.853-0.859,6.567-2.566    c1.713-1.718,2.568-3.901,2.568-6.57v-91.358c0-2.662-0.855-4.853-2.568-6.57C169.306,366.307,167.116,365.451,164.454,365.451z"
                style={
                  Object {
                    "fill": "#ffffff",
                  }
                }
              />
            </svg>
          </div>
          <span
            className="glyphicon glyphicon-ok"
            style={
              Object {
                "color": "var(--reverse-color)",
                "margin": "0px 20px",
                "opacity": 0.2,
              }
            }
            title="icons.no_selections_found"
          />
          <span
            className="glyphicon glyphicon-pencil"
            style={
              Object {
                "color": "var(--reverse-color)",
                "margin": "0px 20px",
                "opacity": 0.2,
              }
            }
            title="icons.no_verse_edits_found"
          />
          <span
            className="glyphicon glyphicon-comment"
            style={
              Object {
                "color": "var(--reverse-color)",
                "margin": "0px 20px",
                "opacity": 0.2,
              }
            }
            title="icons.no_comments_found"
          />
          <span
            className="glyphicon glyphicon-bookmark"
            style={
              Object {
                "color": "var(--reverse-color)",
                "margin": "0px 20px",
                "opacity": 0.2,
              }
            }
            title="icons.not_bookmarked"
          />
        </div>
      </div>
      <div
        className="check-area"
      >
        <div
          style={
            Object {
              "display": "flex",
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <div
            className="verse-title"
          >
            <div
              className="pane"
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                }
              }
            >
              <span
                className="verse-title-title"
              >
                Fwe
              </span>
              <span
                className="verse-title-subtitle"
              >
                Titus
                 
                2:5
              </span>
            </div>
          </div>
          <div>
            <div
              className="ltr-content"
            >
              <div
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
              >
                <span>
                  Kono ewe, oambe ziyenderera ne ntaero zisepahara. 
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "borderLeft": "1px solid var(--border-color)",
              "display": "flex",
              "flex": 1,
              "justifyContent": "center",
              "overflowY": "auto",
            }
          }
        >
          <div
            style={
              Object {
                "WebkitUserSelect": "none",
                "alignItems": "center",
                "display": "flex",
                "flex": "1",
                "justifyContent": "center",
                "overflow": "auto",
              }
            }
          >
            <div
              className="instructions-area"
            >
              <div>
                <span>
                  selection_invalidated
                  <strong
                    data-class="selection-tooltip"
                    data-delay-hide="100"
                    data-effect="float"
                    data-place="top"
                    data-tip="invalidated_tooltip"
                    data-type="dark"
                    style={
                      Object {
                        "font-size": "0.8em",
                        "vertical-align": "super",
                      }
                    }
                  >
                    1
                  </strong>
                </span>
                <div
                  className="__react_component_tooltip place-top type-dark "
                  data-id="tooltip"
                  id={undefined}
                />
              </div>
              <span>
                please_select
              </span>
              <br />
              <span>
                <strong
                  style={
                    Object {
                      "color": "var(--accent-color)",
                    }
                  }
                >
                  ""
                </strong>
              </span>
              <br />
            </div>
          </div>
        </div>
      </div>
      <div
        className="actions-area"
      >
        <button
          className="btn-second"
          onClick={[Function]}
          style={
            Object {
              "alignSelf": "flex-start",
            }
          }
        >
          cancel
        </button>
        <button
          className="btn-second"
          disabled={true}
          onClick={[Function]}
        >
          <span
            className="glyphicon glyphicon-erase"
            style={
              Object {
                "marginRight": "10px",
              }
            }
          />
          clear_selection
        </button>
        <button
          className="btn-prime"
          disabled={true}
          onClick={[Function]}
        >
          <span
            className="glyphicon glyphicon-ok"
            style={
              Object {
                "marginRight": "10px",
              }
            }
          />
          save
        </button>
      </div>
    </div>
    <div />
  </div>
  <div />
</div>
`;

exports[`View component Tests Integrated View test with verseEdit 1`] = `
<div
  className="verse-check"
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <div
      className="verse-check-card"
    >
      <div
        className="title-bar"
      >
        <span>
          select
        </span>
        <div
          style={
            Object {
              "display": "flex",
              "justifyContent": "flex-end",
            }
          }
        >
          <span
            className="glyphicon glyphicon-ok"
            style={
              Object {
                "color": "var(--reverse-color)",
                "margin": "0px 20px",
                "opacity": 0.2,
              }
            }
            title="icons.no_selections_found"
          />
          <span
            className="glyphicon glyphicon-pencil"
            style={
              Object {
                "color": "var(--reverse-color)",
                "margin": "0px 20px",
                "opacity": 1,
              }
            }
            title="icons.verse_edits_found"
          />
          <span
            className="glyphicon glyphicon-comment"
            style={
              Object {
                "color": "var(--reverse-color)",
                "margin": "0px 20px",
                "opacity": 0.2,
              }
            }
            title="icons.no_comments_found"
          />
          <span
            className="glyphicon glyphicon-bookmark"
            style={
              Object {
                "color": "var(--reverse-color)",
                "margin": "0px 20px",
                "opacity": 0.2,
              }
            }
            title="icons.not_bookmarked"
          />
        </div>
      </div>
      <div
        className="check-area"
      >
        <div
          style={
            Object {
              "display": "flex",
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <div
            className="verse-title"
          >
            <div
              className="pane"
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                }
              }
            >
              <span
                className="verse-title-title"
              >
                Fwe
              </span>
              <span
                className="verse-title-subtitle"
              >
                Titus
                 
                2:5
              </span>
            </div>
          </div>
          <div>
            <div
              className="ltr-content"
            >
              <div
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
              >
                <span>
                  Kono ewe, oambe ziyenderera ne ntaero zisepahara. 
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "borderLeft": "1px solid var(--border-color)",
              "display": "flex",
              "flex": 1,
              "justifyContent": "center",
              "overflowY": "auto",
            }
          }
        >
          <div
            style={
              Object {
                "WebkitUserSelect": "none",
                "alignItems": "center",
                "display": "flex",
                "flex": "1",
                "justifyContent": "center",
                "overflow": "auto",
              }
            }
          >
            <div
              className="instructions-area"
            >
              <span>
                please_select
              </span>
              <br />
              <span>
                <strong
                  style={
                    Object {
                      "color": "var(--accent-color)",
                    }
                  }
                >
                  ""
                </strong>
              </span>
              <br />
            </div>
          </div>
        </div>
      </div>
      <div
        className="actions-area"
      >
        <button
          className="btn-second"
          onClick={[Function]}
          style={
            Object {
              "alignSelf": "flex-start",
            }
          }
        >
          cancel
        </button>
        <button
          className="btn-second"
          disabled={true}
          onClick={[Function]}
        >
          <span
            className="glyphicon glyphicon-erase"
            style={
              Object {
                "marginRight": "10px",
              }
            }
          />
          clear_selection
        </button>
        <button
          className="btn-prime"
          disabled={true}
          onClick={[Function]}
        >
          <span
            className="glyphicon glyphicon-ok"
            style={
              Object {
                "marginRight": "10px",
              }
            }
          />
          save
        </button>
      </div>
    </div>
    <div />
  </div>
  <div />
</div>
`;
